module/models/action.mjs

```diff
@@ -1762,6 +1782,9 @@ export default class CrucibleAction extends foundry.abstract.DataModel {
       outcome.rolls = outcomeRolls.map(i => message.rolls[i]);
       action.outcomes.set(outcome.target, outcome);
     }
+
+    // Add reference to originating message
+    action.message = message;
```

Comment:
Let's pass the `message` in as part of `actionContext` to the constructor instead of assigning a reference post-construction and post-preparation.